flink-formats/flink-protobuf/src/main/java/org/apache/flink/formats/protobuf/PbSchemaValidator.java

```diff
@@ -0,0 +1,189 @@
+ */
+
+package org.apache.flink.formats.protobuf;
+
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.types.logical.ArrayType;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.LogicalTypeRoot;
+import org.apache.flink.table.types.logical.MapType;
+import org.apache.flink.table.types.logical.RowType;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.Descriptors.FieldDescriptor.JavaType;
+
+import java.util.EnumMap;
+import java.util.EnumSet;
+
+/** Validation class to verify protobuf definition and flink schema. */
+public class PbSchemaValidator {
+    private Descriptors.Descriptor descriptor;
+    private RowType rowType;
+
+    private static final EnumMap<JavaType, EnumSet<LogicalTypeRoot>> TYPE_MATCH_MAP =
+            new EnumMap(JavaType.class);
+
+    static {
+        TYPE_MATCH_MAP.put(JavaType.BOOLEAN, EnumSet.of(LogicalTypeRoot.BOOLEAN));
+        TYPE_MATCH_MAP.put(
+                JavaType.BYTE_STRING,
+                EnumSet.of(LogicalTypeRoot.BINARY, LogicalTypeRoot.VARBINARY));
+        TYPE_MATCH_MAP.put(JavaType.DOUBLE, EnumSet.of(LogicalTypeRoot.DOUBLE));
+        TYPE_MATCH_MAP.put(JavaType.FLOAT, EnumSet.of(LogicalTypeRoot.FLOAT));
+        TYPE_MATCH_MAP.put(
+                JavaType.ENUM,
+                EnumSet.of(
+                        LogicalTypeRoot.VARCHAR,
+                        LogicalTypeRoot.CHAR,
+                        LogicalTypeRoot.TINYINT,
+                        LogicalTypeRoot.SMALLINT,
+                        LogicalTypeRoot.INTEGER,
+                        LogicalTypeRoot.BIGINT));
+        TYPE_MATCH_MAP.put(
+                JavaType.STRING, EnumSet.of(LogicalTypeRoot.VARCHAR, LogicalTypeRoot.CHAR));
+        TYPE_MATCH_MAP.put(JavaType.INT, EnumSet.of(LogicalTypeRoot.INTEGER));
+        TYPE_MATCH_MAP.put(JavaType.LONG, EnumSet.of(LogicalTypeRoot.BIGINT));
+    }
+
+    public PbSchemaValidator(Descriptors.Descriptor descriptor, RowType rowType) {
+        this.descriptor = descriptor;
+        this.rowType = rowType;
+    }
+
+    public Descriptors.Descriptor getDescriptor() {
+        return descriptor;
+    }
+
+    public void setDescriptor(Descriptors.Descriptor descriptor) {
+        this.descriptor = descriptor;
+    }
+
+    public RowType getRowType() {
+        return rowType;
+    }
+
+    public void setRowType(RowType rowType) {
+        this.rowType = rowType;
+    }
+
+    public void validate() {
+        validateTypeMatch(descriptor, rowType);
+    }
+
+    /**
+     * Validate type match of row type.
+     *
+     * @param descriptor the {@link Descriptors.Descriptor} of the protobuf object.
+     * @param rowType the corresponding {@link RowType} to the {@link Descriptors.Descriptor}
+     */
+    public void validateTypeMatch(Descriptors.Descriptor descriptor, RowType rowType) {
```

Comment:
`public` -> `private`.  
Another general rule, we'd better to make the method `private` as much as possible.